src/library_webgpu.js

```diff
@@ -2565,6 +2577,9 @@ var LibraryWebGPU = {
       callUserCallback(() => {
         var deviceWrapper = { queueId: WebGPU.mgrQueue.create(device["queue"]) };
         var deviceId = WebGPU.mgrDevice.create(device, deviceWrapper);
+        // Register device lost lisenter
+        deviceWrapper.lostCallback = desc["deviceLostCallback"];
+        deviceWrapper.object["lost"].then((info) => deviceWrapper.lostCallback(info));
```

Comment:
We have `SetDeviceLostCallback` removed and moved to device descriptor. I'm really unsure if I update this piece correct

Reply:
deviceWrapper.lostCallback isn't needed anymore, I'm pretty sure that was just for handling multiple SetDeviceLostCallback calls which isn't possible anymore. I think you can do more simply:
```
if (deviceLostCallbackPtr) {
  device["lost"].then(info => callUserCallback(() => WebGPU.errorCallback(...)));
}
```

Reply:
It looks correct though.

Reply:
I realize this is a couple of months late, but can someone expand a bit on why lostCallback isn't needed anymore? i.e., is that class of error no longer possible? Or is it reported in a different way? (and if so, does client code need to proactively add/change code to get the reporting?)

Reply:
@steven-johnson this is just an internal implementation detail. The relevant change to the C API is that you have to set the lostCallback at device creation rather than being able to do it later on.